### DISCUSSION TITLE
FastAPI endpoint downloading incorrect content from file URL

### DISCUSSION THREAD

**Discussion by kaan9700** [fastapi/fastapi#11806] [Questions]
### First Check

- [x] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.post("/upload-file/")
def upload_file(request: UploadFileRequest):
    # Download the file from the URL
    local_filename = f"{MEDIA_DIR}/{request.fileName}"
    url = request.fileURL
    print(f"Downloading file from {url}")

    username = 'test@liferay.com'
    password = 'test1234'

    response = requests.get(url, auth=HTTPBasicAuth(username, password))
    
    if response.status_code == 200:
        with open('GETTING_STARTED.html', 'wb') as file:
            file.write(response.content)
        print("File successfully downloaded")
    else:
        print(f"Error: {response.status_code}")
```

### Description

I have a FastAPI backend running locally on port 8000. I also have Liferay running locally via Docker. Files can be uploaded via Liferay. The upload triggers an event that sends the URL of the file to my FastAPI backend to move the file there.

I first tried using a standalone Python script to make a GET request to the URL to get the file, which worked fine. The content of the file is complete and correct.

I then implemented the same code as a function in my FastAPI endpoint (which is called by Liferay). The file is downloaded, but the content is completely different. The content is an HTML page. When I opened the HTML document I saw that I landed on the login page of Liferay. 

The only difference between the two methods is the request header.

For the standalone script it looks like this:
{'User-Agent': 'python-requests/2.31.0', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'}

For the endpoint it looks like this:
{'User-Agent': 'python-requests/2.31.0', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive', 'Cookie': 'JSESSIONID=FDFC0EB495BE73045A5383EFAD88E0B4; COOKIE_SUPPORT=true; GUEST_LANGUAGE_ID=en_US'}

Is it possible to omit the cookies in a request from FastAPI? Or does anyone else know the reason for this?

### Operating System

Windows

### FastAPI Version

0.110.2

### Pydantic Version

2.7.1

### Python Version

Python 3.10.13

**Comment by gustavosett** [comment]
It's likely because the session cookies are leading you back to the login page instead of letting you access the file directly. 

1. When you make the GET request within FastAPI, make sure to not forward the incoming request cookies to the file server. This can be done by explicitly setting the `cookies` parameter to an empty dictionary in your `requests.get` call.
   
2. If necessary, manually set the headers to mimic the standalone script, particularly excluding any session or user-specific cookies.

Try something like this:

```python
from requests.auth import HTTPBasicAuth
import requests

@app.post("/upload-file/")
def upload_file(request: UploadFileRequest):
    local_filename = f"{MEDIA_DIR}/{request.fileName}"
    url = request.fileURL
    print(f"Downloading file from {url}")

    username = 'test@liferay.com'
    password = 'test1234'

    # Make the request without cookies
    response = requests.get(url, auth=HTTPBasicAuth(username, password), cookies={})

    if response.status_code == 200:
        with open(local_filename, 'wb') as file:
            file.write(response.content)
        print("File successfully downloaded")
    else:
        print(f"Error: {response.status_code}")
```

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
